### PR TITLE
ci: fix some failing long-running tests related to password requirements

### PR DESCRIPTION
**Pull request by jesse-amano-hpe**
## Description

Addressing CircleCI test failures on Slurm clusters, Debian package install, and e2e "local" and perf tests.

## Test Plan

CI tests pass

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by stoksc — COMMENTED** [review]
i see znode tests are still failing too. i'm running https://app.circleci.com/pipelines/github/determined-ai/determined/56052/workflows/4c820a30-5f45-407c-b11b-115cfc9aead6 to see if it fixes it. if it does i can merge into your branch.

**Review by stoksc — APPROVED** [review]

File: .circleci/real_config.yml [hunk 1]
```diff
@@ -1932,6 +1933,11 @@ jobs:
       - install-devcluster
       - start-devcluster:
           target-stage: db
+      - run: |
+          sudo mkdir -p /etc/systemd/system/determined-master.service.d
+          echo "[Service]" | sudo tee /etc/systemd/system/determined-master.service.d/password.override.conf >/dev/null
```
[the remark this answers is not in the document]
Comment (jesse-amano-hpe) [reply .circleci/real_config.yml:1938]: Shell pipes and redirects (like `>>`) create new processes, but there isn't a way to apply a temporary owner to them; `sudo echo narf >> poit` runs `echo` as a superuser, but opens the `poit` file with the current session's user, which in this case wouldn't have permission to write to /etc/systemd

**Review by loksonarius — APPROVED** [review]

File: .circleci/real_config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dannysauer) [reply .circleci/real_config.yml:1945]: This is going to fail in weird ways if sudo is configured to require a password.

But more importantly, why not do the whole file in one block so there's less potential "use `-a` except on the first line" confusion?  Either create it as a template file and `sed` the password into the template while outputting to the override location, or perhaps printf so you don't have nested quote problems?

```
printf '[Service]\nEnvironment="DET_SECURITY_INITIAL_USER_PASSWORD=%s"\n' "$INITIAL_USER_PASSWORD" \
| sudo tee /etc/systemd/system/determined-master.service.d/password.override.conf > /dev/null
```

File: .circleci/real_config.yml [hunk 2]
```diff
@@ -2643,6 +2649,11 @@ jobs:
               echo "export OPT_DEVBOX_PREFIX=circleci-job-$(echo -n "${CIRCLE_USERNAME}-${CIRCLE_BRANCH}-${CIRCLE_JOB}" | md5sum | awk '{print $1}')" >> "$BASH_ENV"
             fi
 
+      - run:
+          name: Set initial user password
+          command: |
+            echo "export INITIAL_USER_PASSWORD=${INITIAL_USER_PASSWORD}" >> "$BASH_ENV"
```
[the remark this answers is not in the document]
Comment (dannysauer) [reply .circleci/real_config.yml:2706]: Sets the variable in the environment of future commands run within the same job.

**Review by dannysauer — APPROVED** [review]
This looks fine to me from an infra perspective.  My only note is that the variable and config refers to initial "user" password - is that setting the admin and determined user both to the same initial password?  I don't remember how that works. :)  As long as that's the case, I'm happy.  My other in-line note is more trivial style opinion than anything functional.

**Comment by jesse-amano-hpe** [comment]
> This looks fine to me from an infra perspective. My only note is that the variable and config refers to initial "user" password - is that setting the admin and determined user both to the same initial password? I don't remember how that works. :) As long as that's the case, I'm happy. My other in-line note is more trivial style opinion than anything functional.

Yes, it sets the admin and determined user passwords to the same value.

File: .circleci/real_config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jesse-amano-hpe) [reply .circleci/real_config.yml:1945]: Ooh, ...drat, wish I'd thought of that. I'll give that a try when I can and maybe patch it in as a separate very-smol PR

**Pull request merged**